src/pages/basic_data/app_admin/CustomerDoiPasswordField.tsx

```diff
@@ -1,91 +1,113 @@
-                  <Tooltip
-                    title={
-                      showPassword
-                        ? t('basic_data.institutions.password_hide')
-                        : t('basic_data.institutions.password_show')
-                    }>
-                    <IconButton onClick={() => setShowPassword(!showPassword)}>
-                      {showPassword ? <VisibilityIcon /> : <VisibilityOffIcon />}
-                    </IconButton>
-                  </Tooltip>
-                ),
-              }}
-            />
-          )}
-        </Field>
-      )}
+      <LoadingButton
+        disabled={disabled}
+        sx={{ height: 'fit-content', minWidth: '7rem', alignSelf: 'center' }}
+        variant="outlined"
+        loading={doiAgentQuery.isLoading && showPasswordInputModal}
+        onClick={() => setShowPasswordInputModal(true)}>
+        {t('basic_data.institutions.doi_password')}
+      </LoadingButton>
+
+      <Modal title={'Passord'} open={showPasswordInputModal} onClose={cancelPasswordChange}>
+        <DialogContent>
+          <Field
+            name={CustomerInstitutionFieldNames.DoiPassword}
+            validate={(value: string) =>
+              value
+                ? undefined
+                : t('feedback.validation.is_required', {
+                    field: t('basic_data.institutions.doi_password'),
+                  })
+            }>
+            {({ field, meta: { touched, error } }: FieldProps<string>) => (
+              <TextField
+                {...field}
+                value={field.value ?? ''}
+                data-testid={dataTestId.basicData.institutionAdmin.doiPasswordField}
+                label={t('basic_data.institutions.doi_password')}
+                disabled={disabled}
+                required={!disabled}
+                fullWidth
+                type={showPassword ? 'text' : 'password'}
+                variant="filled"
+                error={touched && !!error}
+                helperText={<ErrorMessage name={field.name} />}
+                InputProps={{
+                  endAdornment: (
+                    <Tooltip
+                      title={
+                        showPassword
+                          ? t('basic_data.institutions.password_hide')
+                          : t('basic_data.institutions.password_show')
+                      }>
+                      <IconButton onClick={() => setShowPassword(!showPassword)}>
+                        {showPassword ? <VisibilityIcon /> : <VisibilityOffIcon />}
+                      </IconButton>
+                    </Tooltip>
+                  ),
+                }}
+              />
+            )}
+          </Field>
+        </DialogContent>
+        <DialogActions>
+          <Button data-testid="cancel-update-doi-password" onClick={cancelPasswordChange}>
+            {t('common.cancel')}
+          </Button>
+          <Button
+            data-testid="save-update-doi-password"
```

Comment:
legg data-testids her til `dataTestIds`, sånn at Eirik også kan bruke dem

Reply:
kunne godt lagt det til `common.save` og `common.cancel` tror jeg, så kan vi bruke dem mer fremover, i stedet for å ha mange ulike IDer for lagring/avbryting